app/src/main/java/com/amaze/filemanager/ui/ItemPopupMenu.java

```diff
@@ -157,8 +157,19 @@ public boolean onMenuItemClick(MenuItem item) {
       case R.id.delete:
         ArrayList<LayoutElementParcelable> positions = new ArrayList<>();
         positions.add(rowItem);
-        GeneralDialogCreation.deleteFilesDialog(
-            context, mainActivity, positions, utilitiesProvider.getAppTheme());
+        SharedPreferences sharedPreferences =
+            PreferenceManager.getDefaultSharedPreferences(context);
+        boolean deletePermanently =
+            sharedPreferences.getBoolean(
+                PreferencesConstants.PREFERENCE_DELETE_PERMANENTLY_WITHOUT_CONFIRMATION,
+                PreferencesConstants.DEFAULT_PREFERENCE_DELETE_PERMANENTLY_WITHOUT_CONFIRMATION);
```

Comment:
please extract this to a function, probably inside mainactivityhelper class

Reply:
some more changes:

- please add the info that this disables the trashbin to the description.
- also, we should still show the confirmation dialog if "delete confirmation" is checked. the current implementation essentially revokes that previous setting!

thanks!